cmd/libs/go2idl/deepcopy-gen/generators/deepcopy.go

```diff
@@ -134,17 +150,36 @@ func (g *genDeepCopy) Namers(c *generator.Context) namer.NameSystems {
 
 func (g *genDeepCopy) Filter(c *generator.Context, t *types.Type) bool {
 	// Filter out all types not copyable within the package.
-	copyable := copyableWithinPackage(t)
+	copyable := g.copyableWithinPackage(t)
 	if copyable {
 		g.typesForInit = append(g.typesForInit, t)
 	}
 	return copyable
 }
 
-func copyableWithinPackage(t *types.Type) bool {
+func (g *genDeepCopy) copyableWithinPackage(t *types.Type) bool {
+	return copyableWithinPackage(t, g.boundingDirs)
+}
+
+func isRootedUnder(pkg string, roots []string) bool {
+	// Add trailing / to avoid false matches, e.g. foo/bar vs foo/barn.
+	pkg = pkg + "/"
```

Comment:
roots will not have the trailing slash, and this does a prefix check so it becomes `strings.HasPrefix("/foo/bar//", "foo/")` which still matches.  Am I messing something up?